source/common/network/io_socket_handle_impl.cc

```diff
@@ -71,7 +72,10 @@ Api::IoCallUint64Result IoSocketHandleImpl::sendmsg(const Buffer::RawSlice* slic
                                                     const Address::Instance& peer_address) {
   const auto* address_base = dynamic_cast<const Address::InstanceBase*>(&peer_address);
   sockaddr* sock_addr = const_cast<sockaddr*>(address_base->sockAddr());
-
+  if (sock_addr == nullptr) {
+    // Unlikely to happen unless the wrong peer address is passed.
```

Comment:
Do you want to leave this in and test it, or revert and just rely on the audit TODO I suggested above?

Reply:
We can leave this: since it is covered by test case UdpSendToInternalAddressWithSocketInterface